### ISSUE TITLE
`--outputjson` line numbers are off by one

### ISSUE THREAD

**Issue by kevin-hanselman** [microsoft/pyright#8760]
**Describe the bug**
The line numbers reported in pyright's JSON output do not match those reported in the standard output. 

**Code or Screenshots**
If possible, provide a minimal, self-contained code sample (surrounded by triple back ticks) to demonstrate the issue. The code should define or import all referenced symbols.

A minimal example:

Consider a file called `pyright_bug.py`:

```python
def add_one(x: int) -> int:
    return x + 1

add_one(None)
```

After running `pyright pyright_bug.py`:

```
.../pyright_bug.py
  .../pyright_bug.py:4:9 - error: Argument of type "None" cannot be assigned to parameter "x" of type "int" in function "add_one"
    "None" is incompatible with "int" (reportArgumentType)
1 error, 0 warnings, 0 informations
```

Note the output says that the error is on line 4. I agree with this output.

After running `pyright --outputjson pyright_bug.py`:

```json
{
    "version": "1.1.376",
    "time": "1723653981091",
    "generalDiagnostics": [
        {
            "file": ".../pyright_bug.py",
            "severity": "error",
            "message": "Argument of type \"None\" cannot be assigned to parameter \"x\" of type \"int\" in function \"add_one\"\n  \"None\" is incompatible with \"int\"",
            "range": {
                "start": {
                    "line": 3,
                    "character": 8
                },
                "end": {
                    "line": 3,
                    "character": 12
                }
            },
            "rule": "reportArgumentType"
        }
    ],
    "summary": {
        "filesAnalyzed": 1,
        "errorCount": 1,
        "warningCount": 0,
        "informationCount": 0,
        "timeInSec": 0.125
    }
}
```

Note this JSON output says that the error is on line 3.

Is this intentional? Does the JSON output use zero-based indexing?

**Comment by erictraut** [comment]
Yes, this is intentional. Both line and character offsets are zero-based as indicated in the [documentation](https://microsoft.github.io/pyright/#/command-line?id=json-output).
